src/modm/driver/inertial/ixm42xxx.hpp

```diff
@@ -0,0 +1,200 @@
+    readAccelData();
+
+    /**
+     * @brief Read the gyro data from the device
+     * @return False in case of any error, e.g. if some register acces is not
+     * permitted.
+     */
+    modm::ResumableResult<bool>
+    readGyroData();
+
+    /**
+     * @brief Read the sensor data from the device
+     * @return False in case of any error, e.g. if some register acces is not
+     * permitted.
+     */
+    modm::ResumableResult<bool>
+    readSensorData();
+
+    /**
+     * @brief Read the FSYNC timestamp from the device
+     * @return False in case of any error, e.g. if some register acces is not
+     * permitted.
+     */
+    modm::ResumableResult<bool>
+    readFsyncTimestamp(uint16_t &timestamp);
+
+    /**
+     * @brief Read the FIFO count from the device
+     * @return False in case of any error, e.g. if some register acces is not
+     * permitted.
+     */
+    modm::ResumableResult<bool>
+    readFifoCount(uint16_t &count);
+
+    /**
+     * @brief Read the FIFO data from the device
+     * @return False in case of any error, e.g. if some register acces is not
+     * permitted.
+     */
+    modm::ResumableResult<bool>
+    readFifoData();
+
+    /**
+     * @brief Set the FIFO watermark used to generate FIFO_WM_GT interrupt
+     * @warning The FIFO watermarkl should be set, before choosing this interrupt source.
+     * @return False in case of any error, e.g. if some register acces is not
+     * permitted.
+     */
+    modm::ResumableResult<bool>
+    writeFifoWatermark(uint16_t watermark);
+
+public:
+    /**
+     * @brief update a single register.
+     *
+     * @param reg The register to be updated
+     * @param setMask The bits to be setted in the register
+     * @param clearMask The bits to be cleared in the register.
+     * @return False in case of any error, e.g. if some register acces is not
+     * permitted.
+     *
+     * @warning Only the registers GYRO_CONFIG0, ACCEL_CONFIG0 and PWR_MGMT0
+     * can be modified during sensor operation
+     */
+    modm::ResumableResult<bool>
+    updateRegister(Register reg, Register_t setMask, Register_t clearMask = Register_t(0xff));
+
+    /**
+     * @brief Write a single register.
+     *
+     * @param reg The register to be read
+     * @param value The value to write to the register
+     * @return False in case of any error, e.g. if some register acces is not
+     * permitted.
+     *
+     * @warning Only the registers GYRO_CONFIG0, ACCEL_CONFIG0 and PWR_MGMT0
+     * can be modified during sensor operation
+     */
+    modm::ResumableResult<bool>
+    writeRegister(Register reg, uint8_t value);
+
+    /**
+     * @brief Read a single register.
+     *
+     * @param reg The register to be read
+     * @param value The placeholder for the read value to be stored
+     * @return False in case of any error, e.g. if some register acces is not
+     * permitted.
+     */
+    modm::ResumableResult<bool>
+    readRegister(Register reg, uint8_t &value);
```

Comment:
```suggestion
    readRegister(Register reg, uint8_t *value);
```
Makes it clearer at the call site that value will be modified.

Reply:
I have changed this, but I believe it is a bit inconsistent now since many other "getter" methods use the reference e.g. readFifoCount.

Reply:
Should we use pointers in this one instance?

Reply:
Sorry, yes, ideally we should always use pointers for arguments that get written to make it clear at the callsite that the variable gets written.
Unfortunately we messed it up with our Uart::read() interface function and it's been getting copied ever since. But let's do it right now, so I don't have to refactor it later.

Reply:
Are you ok with changing all the &args to *args also in the data class?

Reply:
I have done this now. Should Data still be passed to Ixm42xxx with reference?

Reply:
You can make the Data object a member variable of the driver. The separate passing of the data object was only for packing a bunch of data objects into a packed struct and sending that over the wire as an attempt to serialize very cheaply. However, this only worked for a specific use case and is otherwise kinda dumb and annoying.

Reply:
The only problem with making it a member variable is that one should specify if it is Data or FifoData and provide the fifo buffer size if it is FifoData. In that case the driver should take the fifo buffer size as template argument

Reply:
Yeah right. Let's just leave it as it is, and I'll survey all drivers in the future and come up with a proper solution for this.